new_device.py

```diff
@@ -0,0 +1,171 @@
+import torch
+from torch.overrides import enable_torch_function_mode, TorchFunctionMode
+from torch.utils._pytree import tree_map
+
+import numpy as np
+
+aten = torch.ops.aten
+
+# 1. A Tensor that stores custom raw_data and implement functions for it
+class MyDeviceTensor(torch.Tensor):
+    IMPLEMENTATIONS = {}
+
+    @staticmethod
+    def __new__(cls, size, dtype, raw_data=None, requires_grad=False):
+        # Use a meta Tensor here to be used as the wrapper
+        return torch.Tensor._make_subclass(cls, torch.empty(size, dtype=dtype, device="meta"), require_grad=requires_grad)
+
+    def __init__(self, size, dtype, raw_data=None, requires_grad=False):
+        # Store any provided user raw_data
+        self.raw_data = raw_data
+
+    def __repr__(self):
+        st = super().__repr__()
+        st = st.replace("device='meta'", "device='my_device'")
```

Comment:
This looks more confusing than hell ha ha ha

Reply:
Yes, this most likely will need to be tweaked by real backends. This is just to be able to easily re-use the plain Tensor print.